api/v1alpha1/operator_types.go

```diff
@@ -27,6 +27,11 @@ type OperatorSpec struct {
 	//+kubebuilder:validation:MaxLength:=48
 	//+kubebuilder:validation:Pattern:=^[a-z0-9]+(-[a-z0-9]+)*$
 	PackageName string `json:"packageName"`
+
+	//+kubebuilder:validation:MaxLength:=64
+	//+kubebuilder:validation:Pattern:=^(\|\||\s+)?([\s~^><=]*)v?(\d+)(\.(\d+))?(\.(\d+))?(\-(.+))?$
```

Comment:
Will this match the full set of (but not more than) version ranges supported by `semver.ParseRange`?

Reply:
AFACT, this doesn't support the full syntax.

Reply:
The semver package explicitly mentions no regex... Where did that one come from?

Reply:
https://ihateregex.io/expr/semver/ then add the comparison operators

Reply:
Either way, it's not correct as-is, and needs to account for the range prefix elements.

Reply:
Well, based on the milestone 3 consensus, it sounds like we're going with `spec.version` and a single explicit version for now. So I think we can literally use the linked regex?

Reply:
Based on our upstream convo yesterday, I've left the name as "version" (I'm also happy with PackageVersion, if that's prefered - I personally like the succinctness of `version`. I've added some comments with examples and a link to the semver.org site.